nixos/modules/programs/gnupg.nix

```diff
@@ -66,16 +64,17 @@ in
     };
 
     agent.pinentryFlavor = mkOption {
-      type = types.nullOr (types.enum pkgs.pinentry.flavors);
+      # keep in sync with all possible favlors of pinentry
+      type = types.nullOr (types.enum [ "curses" "tty" "emacs" "gtk2" "gnome3" "qt" ]);
```

Comment:
If we're changing this type, why not also allow packages ([similar to home-manager's `rbw` module](https://github.com/nix-community/home-manager/blob/4d8f90205c6c90be2e81d94d0e5eedf71c1ba34e/modules/programs/rbw.nix#L44)).

This would allow using other `pinentry` implementations, like [`pinentry-rofi`](https://github.com/plattfot/pinentry-rofi).

Reply:
Please open another PR with that new feature. I don't want to expand the scope of this PR even further.